chain/gen/genesis/util.go

```diff
@@ -46,3 +49,14 @@ func doExecValue(ctx context.Context, vm *vm.VM, to, from address.Address, value
 
 	return ret.Return, nil
 }
+
+var GenesisNetworkVersion = func() network.Version { // TODO: Get from build/
+	if build.UseNewestNetwork() { // TODO: Get from build/
+		return build.NewestNetworkVersion // TODO: Get from build/
+	} // TODO: Get from build/
+	return network.Version1 // TODO: Get from build/
+}() // TODO: Get from build/
+
+func genesisNetworkVersion(context.Context, abi.ChainEpoch) network.Version { // TODO: Get from build/
+	return GenesisNetworkVersion // TODO: Get from build/
+} // TODO: Get from build/
```

Comment:
Much todo

Though it looks like the genesis code is fairly hard-coded at v0, and we'll need some way to create networks with newer actor versions.

Really I think that we should do one of those 2 things:
* Move genesis construction to actors, which gets us versioning for free
* Keep things at v0, run all updates just after creating the genesis (maybe fine?)

Reply:
Honestly, I think this logic belongs in actors.